runtime/php/compiler.Dockerfile

```
@@ -19,15 +19,17 @@ LABEL home="https://github.com/brefphp/bref"
 WORKDIR /tmp
 
 
-# Lambda is based on 2017.03. Lock YUM to that release version.
-RUN sed -i 's/releasever=latest/releaserver=2017.03/' /etc/yum.conf
+# Lambda is based on 2018.03. Lock YUM to that release version.
+RUN sed -i 's/releasever=latest/releaserver=2018.03/' /etc/yum.conf
 
 
 RUN set -xe \
 # Download yum repository data to cache
  && yum makecache \
 # Default Development Tools
- && yum groupinstall -y "Development Tools"  --setopt=group_package_types=mandatory,default
+ && yum groupinstall -y "Development Tools"  --setopt=group_package_types=mandatory,default \
+# Upgrade gcc
+ && yum install -y gcc72 gcc72-c++
```

I wonder if this could simply be

```suggestion
 && yum install -y gcc72-c++
```

since only `gcc72-c++` is missing.

---

I though about it too, my main concern is a future change. Since gcc72 is not directly installed by `Development Tools` but pulled in because of `kernel-devel`, if in the future they change something (unluckily) we might end up with the reverse problem of having gcc48 and gcc72-c++. 

This at least sets a "base level" for the compilers, even in case they drop gcc72 dependency, we would still be using 7.2. An upgrade to 8 or 9 would really be unexpected.

At the end, it's in the build process so someone would catch it anyway and it would not break end users features, but he might end wasting time trying to figure out the error from scratch. I think this might be fail-safe(er) and quite self-explanatory in case of future compiler changes.

---

Understood.

> Since gcc72 is not directly installed by Development Tools but pulled in because of kernel-devel

That's a good point, let's leave it there.